freshmaker/pyxis.py

```diff
@@ -215,6 +215,121 @@ def get_images_by_digests(self, digests):
         :return: bundle ContainerImages
         :rtype: list
         """
-        request_params = {'include': 'data.brew',
-                          'filter': f"repositories.manifest_list_digest=in=({','.join(digests)})"}
-        return self._pagination("images", request_params)
+        request_params = {'include': 'data.brew,data.repositories',
+                          'filter': f'repositories.manifest_list_digest=in=({",".join(digests)})'}
+        return self._pagination('images', request_params)
+
+    def _add_repositories_info(self, reg_repo_info):
+        """
+        For every pair of registry-repository add information about it's
+        auto_rebuild tags. To decrease amount of queries to Pyxis, only one
+        query is performed with the filter set to proper registry-repository
+        pair.
+
+        A list of tags for each repository will be added to the 'tags' key in
+        the input info about the repository.
+
+        :param dict reg_repo_info: map of pairs (registry, repository) to a
+            dict containing nvrs of bundle images from that repo and
+            auto_rebuild tags of that repo
+        """
+        if not reg_repo_info:
+            return None
+        fltr = ""
+        # Construct filter for future request to Pyxis with registry-repository pairs
+        for reg, repo in reg_repo_info.keys():
+            if not fltr:
+                fltr = f"(registry=={reg};repository=={repo})"
+            else:
+                fltr += f",(registry=={reg};repository=={repo})"
```

Comment:
This reduces code duplication.
```suggestion
            if fltr:
                fltr += ","
            fltr = f"(registry=={reg};repository=={repo})"

```